corehq/motech/repeaters/optionvalue.py

```diff
@@ -10,19 +21,23 @@ def __init__(
         default=NOT_SET,
         choices=None,
         schema=None,
+        coder=None,
     ):
         if schema and default is not self.NOT_SET:
             raise ValueError("default not allowed with schema")
         self.default = default
         self.choices = choices
         self.schema = schema
+        self.coder = coder
 
     def __set_name__(self, owner, name):
         self.name = name
 
     def __get__(self, obj, objtype=None):
         if obj is None:
             return self
+        if self.coder:
+            return self.coder.from_json(obj.options[self.name])
```

Comment:
Should this be moved into `if self.name in obj.options:`
```py
if self.name in obj.options:
    if self.coder:
        return self.coder.from_json(obj.options[self.name])
    return obj.options[self.name]
```
As is it will raise `KeyError` if the option is not present.

Reply:
https://github.com/dimagi/commcare-hq/pull/30348/commits/3079fbb12b9c416608d01dd4c3b5ab9f4f684f67